### PR TITLE
Populate ActivityItem `spender` when able to.

### PR DESCRIPTION
**Pull request by 0xDaedalus**
Closes https://github.com/tallycash/extension/issues/2091

### To Test
- Load a new wallet on version `0.14.7`
- Approve a token you have not previously approved before
- Upgrade to version to this branch
- Open the Activity Page.
- [ ] Activity page should not crash.

### CODE REVIEW

File: background/redux-slices/migrations/to-16.ts [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+              annotation &&
+              annotation.type === "asset-approval" &&
+              (annotation.spenderAddress || annotation.spenderName)
+            ) {
+              const spender = {
+                address: annotation.spenderAddress,
+                network: NETWORK_BY_CHAIN_ID[chainID],
+                annotation: {
+                  nameOnNetwork: {
+                    name: annotation.spenderName,
```
Comment (mhluongo) [inline background/redux-slices/migrations/to-16.ts:88]: If there's no `spenderName` I don't think there should be a `nameOnNetwork`... versus eg

```typescript
annotation: {
    nameOnNetwork: {
        name: undefined,
        network: ETHEREUM
    }
}
```

which I believe this code will do now?

File: background/redux-slices/migrations/to-16.ts [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+                    network: NETWORK_BY_CHAIN_ID[chainID],
+                  },
+                },
+              }
+
+              const { spenderName, spenderAddress, ...newAnnotation } =
```
Comment (mhluongo) [inline background/redux-slices/migrations/to-16.ts:94]: Maybe consider renaming `newAnnotation`? It's not the new annotation, after all. `oldAnnotationProps` perhaps?

**Review by mhluongo — COMMENTED** [review]
Couple questions before I test!

File: background/redux-slices/migrations/to-16.ts (hunk 1 above)
Comment (0xDaedalus) [reply background/redux-slices/migrations/to-16.ts:88]: right you are ✅

**Review by jagodarybacka — APPROVED** [review]
-> `0.14.7` 
-> Approve SUSHI on Polygon 
-> `main` -> activities broken 
-> this branch -> activities fixed

LGTM 🚢

**Pull request merged**